### PR TITLE
added regex support to Properties\ChangePropertyValue, and added Yaml…

### PR DESCRIPTION
**Pull request by nmck257**
…\ChangePropertyValue with feature parity

### CODE REVIEW

**Comment by nmck257** [comment]
PR mostly for visibility since I added an arg to an existing recipe

File: rewrite-yaml/src/main/java/org/openrewrite/yaml/ChangePropertyValue.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    String newValue;
+
+    @Option(displayName = "Old value",
+            required = false,
+            description = "Only change the property value if it matches the configured `oldValue`.")
+    @Nullable
+    String oldValue;
+
+    @Option(displayName = "Regex",
+            description = "Default false. If enabled, `oldValue` will be interepreted as a Regular Expression, and capture group contents will be available in `newValue`",
```
Comment (mike-solomon) [inline rewrite-yaml/src/main/java/org/openrewrite/yaml/ChangePropertyValue.java:52]: @nmck257 If you have some time at some point, I'm not following what this parameter does. Specifically, I don't understand what `capture group contents will be available in newValue` means.

Comment (nmck257) [reply rewrite-yaml/src/main/java/org/openrewrite/yaml/ChangePropertyValue.java:52]: Sure, quick example:
Input text: "hi I'm Nick"
oldValue: "hi I'm (.+)"
newValue: "hello $1"
regex: true
Output text: "hello Nick"

Comment (mike-solomon) [reply rewrite-yaml/src/main/java/org/openrewrite/yaml/ChangePropertyValue.java:52]: Thanks -- appreciate it!

**Pull request merged**